### PR TITLE
imxrt/watchdog : Support watchdog interface

### PR DESCRIPTION
**Pull request by jeongchanKim**
For using watchdog in user context, mapping the imxrt watchdog functions to watchdog drivers.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5323d37690f09aab787b1ba2376b2bec87b67128] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5323d37690f09aab787b1ba2376b2bec87b67128] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+	/* Set to new timeout. */
+	priv->config.timeoutValue = (uint16_t)timeout;
+
+	return OK;
+}
+
+static xcpt_t imxrt_wdog_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler)
+{
+#if 0
```
Comment (jeongarmy) [inline os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c]: Is it not supported?

File: os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+	/* Register the watchdog driver as /dev/wdogX. */
+	if (!watchdog_register(devpath, (FAR struct watchdog_lowerhalf_s *)priv)) {
+		/*
+		 * The actual cause of the failure may have been a failure to allocate
+		 * perhaps a failure to register the timer driver (such as if the
+		 * 'depath' were not unique).  We know here but we return EEXIST to
+		 * indicate the failure (implying the non-unique devpath).
+		 */
+		kmm_free(priv);
+		return -EEXIST;
```
Comment (jeongarmy) [inline os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c]: I think it needs to be set as proper errno.

**Pull request merged**